compiler/rustc_ast/src/token.rs

```diff
@@ -911,7 +912,7 @@ impl NonterminalKind {
             sym::ident => NonterminalKind::Ident,
             sym::lifetime => NonterminalKind::Lifetime,
             sym::literal => NonterminalKind::Literal,
-            sym::meta => NonterminalKind::Meta,
+            sym::meta => NonterminalKind::Meta2021,
```

Comment:
Is this actually needed? The grammar doesn't change, right?

Reply:
I'm not sure...

https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=5ccbe0db996ad629df3df4dc2119e2cd
```rust
#[cfg_attr(not(debug), unsafe(Foo))]
struct Foo;
```

The above is a parse error right now as it expects a path and not a keyword. There were 2 implementations strategies I considered, one of which was defining a macro `unsafe` that created the `Attribute` with the proper unsafety, but given that I would still need syntax changes (to deal with the `unsafe`), I opted to instead change the core `AttrItem` itself to have an `unsafety` field. So I'm not really sure if a) this is the right way of doing things and b) if I need to make the macro changes.